src/w3iProxy/pushProviders/internalPushProvider.ts

```diff
@@ -136,27 +136,31 @@ export default class InternalPushProvider implements W3iPushProvider {
       const clientId = await this.pushClient.core.crypto.getClientId()
       // Retrieving FCM token needs to be client side, outside the service worker.
 
-      const token = await getFirebaseToken()
-
-      const subEvListener = (
-        subEv: PushClientTypes.BaseEventArgs<PushClientTypes.PushResponseEventArgs>
-      ) => {
-        if (subEv.params.subscription?.metadata.url === params.metadata.url) {
-          navigator.serviceWorker.ready.then(registration => {
-            registration.active?.postMessage({
-              type: 'INSTALL_SYMKEY_CLIENT',
-              clientId,
-              topic: subEv.topic,
-              token,
-              symkey: subEv.params.subscription?.symKey
+      try {
+        const token = await getFirebaseToken()
+
+        const subEvListener = (
+          subEv: PushClientTypes.BaseEventArgs<PushClientTypes.PushResponseEventArgs>
+        ) => {
+          if (subEv.params.subscription?.metadata.url === params.metadata.url) {
+            navigator.serviceWorker.ready.then(registration => {
+              registration.active?.postMessage({
+                type: 'INSTALL_SYMKEY_CLIENT',
+                clientId,
+                topic: subEv.topic,
+                token,
+                symkey: subEv.params.subscription?.symKey
+              })
             })
-          })
 
-          this.pushClient?.off('push_subscription', subEvListener)
+            this.pushClient?.off('push_subscription', subEvListener)
+          }
         }
-      }
 
-      this.pushClient.on('push_subscription', subEvListener)
+        this.pushClient.on('push_subscription', subEvListener)
+      } catch (e) {
+        console.error(e)
```

Comment:
Please add a proper error message to prefix the error so we can easily identify it at a later point without needing to dig through stack traces/set breakpoints.

Reply:
Added prefix